### PR TITLE
Features/sso

### PR DESCRIPTION
**Pull request by Convly**
Adds SSO capability

### CODE REVIEW

File: packages/strapi-admin/services/role.js [hunk 1]
```diff
@@ -151,26 +151,32 @@ const count = async (params = {}) => {
...
   }
+};
+
+/**
+ * Delete roles in database if they have no user assigned
+ * @param ids query params to find the roles
+ * @returns {Promise<array>}
+ */
+const deleteByIds = async (ids = []) => {
+  await checkRolesIdForDeletion(ids);
```
Comment (alexandrebodin) [inline packages/strapi-admin/services/role.js:179]: check the error message in the frontend

File: packages/strapi-admin/ee/validation/authentication.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const handleReject = error => Promise.reject(formatYupErrors(error));
+
+const providerOptionsUpdateSchema = yup.object().shape({
+  autoRegister: yup.boolean().required(),
+  defaultRole: yup
+    .strapiID()
+    .required()
+    .test('is-valid-role', 'You must submit a valid default role', roleId => {
+      return strapi.admin.services.role.exists({ id: roleId });
```
Comment (alexandrebodin) [inline packages/strapi-admin/ee/validation/authentication.js:13]: verify test can be a promise ?

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

File: packages/strapi-admin/admin/src/translations/en.json [hunk 3]
```diff
@@ -229,8 +242,11 @@
   "app.components.UpgradePlanModal.text-power": "Unlock the full power",
   "app.components.UpgradePlanModal.text-strapi": "of Strapi by upgrading your plan to the",
   "app.components.Users.MagicLink.connect": "Send this link to the user for them to connect.",
+  "app.components.Users.MagicLink.connect.sso": "Send this link to the user, the first login can be mage via a SSO provider",
```
Comment (sam-pires) [inline packages/strapi-admin/admin/src/translations/en.json:245]:
```suggestion
  "app.components.Users.MagicLink.connect.sso": "Send this link to the user, the first login can be made via a SSO provider",
```

**Comment by sam-pires** [comment]
🎉

**Pull request merged**